File: java/spring-boot-rabbitmq/spring-boot-rabbitmq-subscriber/src/main/java/com/meridal/examples/springbootamqp/service/RecordingService.java
package com.meridal.examples.springbootamqp.service;

import com.meridal.examples.springbootrabbitmq.domain.Recording;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.amqp.rabbit.annotation.RabbitListener;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;

@Service
public class RecordingService {

    private static final Logger LOG = LoggerFactory.getLogger(RecordingService.class);

    private final List<Recording> recordings = new ArrayList<>();

    public List<Recording> getAllRecordings() {
        return recordings;
    }

    @RabbitListener(queues = "${messaging.queue}")
    public void saveRecording(Recording recording) {
        this.recordings.add(recording);
        LOG.info("Received message: " + recording.toString());
    }
}

File: kotlin/kotlin-spring-boot-rabbitmq-subscriber/src/main/kotlin/com/meridal/examples/kotlinspringbootamqp/service/RecordingService.kt
package com.meridal.examples.kotlinspringbootamqp.service

import com.meridal.examples.domain.Recording
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.amqp.rabbit.annotation.RabbitListener
import org.springframework.stereotype.Service

@Service
class RecordingService {

    private val LOG: Logger = LoggerFactory.getLogger(RecordingService::class.java)

    private val recordings: MutableList<Recording> = mutableListOf()

    fun getAllRecordings(): List<Recording> {
        return recordings
    }

    @RabbitListener(queues = ["\${messaging.queue}"])
    fun saveRecording(recording: Recording) {
        recordings.add(recording)
        LOG.info("Received message: $recording")
    }
}